sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -751,7 +830,14 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             }
             finally
             {
-                await Task.WhenAll(tasks).ConfigureAwait(false);
+                // await and clean up the _tasks collection
+                await Task.WhenAll(_tasks.Select(t => t.Task)).ConfigureAwait(false);
+                foreach (var (_, tcs) in _tasks)
```

Comment:
Should we just await tasks in the loop?

Reply:
I don't think so - it would mess up the heartbeat where we can update concurrency.

Reply:
Misunderstood what you were saying here.